dev_scripts/exportTranslation.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable default-case */
+/* eslint-disable array-callback-return */
+/* eslint-disable func-names */
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+/* eslint-disable no-console */
+/* eslint-disable no-useless-escape */
+/* eslint-disable no-shadow */
+/* eslint-disable guard-for-in */
+/* eslint-disable prefer-const */
+/* eslint-disable no-undef */
+/* eslint-disable no-restricted-syntax */
+
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+const fs = require('fs');
+
+const localizationFiles = [];
+let index = 0;
+
+// Define mainLanguage as English (gb)
+const mainLanguage = 'gb';
+
+getLocalizationFiles = () => {
+  // Reads all files in localization files and save them in localizationFiles array
+  fs.readdirSync('././src/localization/').forEach(file => {
+    // Filter to only get JSON files
+    if (file.split('.').pop() === 'json') {
+      const fileName = require(`../src/localization/${file}`);
+      localizationFiles.push(fileName);
+      localizationFiles[index].name = file.replace('.json', '');
+      index += 1;
+    }
+  });
+};
+
+// Gets the language entered by the User (fr, la, gil, tl)
+getSelectedLanguage = () => process.argv[2];
+
+generateCsvData = (mainLanguage, selectedLanguage) => {
+  let valueFound = false;
+  // csvFileData keeps all the data stored in json files
+  let csvData = '';
+  // Goes through all localization files
+  localizationFiles.map(localizationFile => {
```

Comment:
no reason to use map here, forEach would do. map is used when you want to create a new array form existing one: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
I changed all `map` by `forEach`. Just a question regarding this, looks like map is faster than forEach (I read it [here](https://codeburst.io/javascript-map-vs-foreach-f38111822c0f)), shouldn't we take that into consideration too?

Reply:
For array based operations map is faster. I don't see how it can be faster for iterating. The example in the link you've sent mutates array element which is being iterated, it's not how forEach is intended to be used